wrappers/gtimelog_parser.py

```diff
@@ -18,65 +18,129 @@ def __init__(self, config):
                                self.settings.virtual_midnight)
         self.aliases = config.get('aliases', {})
         self.line_format = config.get('line_format', '')
+        self.internal_issue_prefixes = self.parse_internal_issue_prefixes(config)
         if self.line_format == 'categorized':
             self.line_format_str = "category: task description | comment"
             self.delimiter = " "
         else:
             self.line_format_str = "task: description | comment"
             self.delimiter = ":"
 
+    def parse_internal_issue_prefixes(self, config):
+        prefixes = config.get('internal_issue_prefixes', "").split(",")
+        # Cleanup in case of spaces
+        return [
+            prefix.strip() for prefix in prefixes
+        ]
+
     def skip_entry(self, entry):
         if '**' in entry:
             return True
         if entry.strip() in ('arrive', 'arrived', 'start'):
             return True
         return False
 
+    def is_dispatch_entry(self, entry):
+        """Returns whether or not this entry is a dispatch"""
+        return '++' in entry
+
+    def is_internal_entry(self, entry):
+        """Returns whether or not this entry is internal"""
+        issue, description = self.get_issue_description(entry)
+        return any([
+            issue.startswith(ref)
+            for ref in self.internal_issue_prefixes
+        ])
+
+    def get_entries_to_dispatch(self, window):
+        """Returns non internal nor dispatchable nor skippable entries"""
+        return [
+            entry
+            for start, stop, duration, tags, entry in window.all_entries()
+            if not (
+                self.skip_entry(entry)
+                or self.is_dispatch_entry(entry)
+                or self.is_internal_entry(entry)
+            )
+        ]
+
+    def get_duration_to_dispatch(self, window):
+        """Returns the time to dispatch to entries
+        i.e. total amount of dispatchable entries divided byt the quantity of
+        entries to dispatch
+        """
+        entries_to_dispatch = self.get_entries_to_dispatch(window)
+        return sum([
+            int(duration.total_seconds())
+            for start, stop, duration, tags, entry in window.all_entries()
+            if self.is_dispatch_entry(entry)
+        ]) / len(entries_to_dispatch)
```

Comment:
I hate 4 lines one-liners.
Could you please store things in variables, and make it pretty? Thanks!

Reply:
Like this ? I personnaly think it's much heavier to read but it's ok

Reply:
Well, that's really not what I meant.
I'm not against multiple lines list comprehensions.
What I do not like is chained operations (like in the following (very dumb) example)
```python
res = [i for i in is].sort().filter().whatever()
```

I do prefer this
```python
items = [i for i in s]
sorted_items = items.sort()
filtered_items = sorted_items.filter()
whatevered_items = filtered_items.whatever()
```

Which in my opinion is more readable, and allows you to write variable names that document the code :)

In this case, I would have probably written smth like this
```python3
durations = [
    int(duration.total_seconds())
    for start, stop, duration, tags, entry in window.all_entries()
    if self.is_dispatch_entry(entry)
]
total_duration = sum(durations)
return total_duration / len(entries_to_dispatch)
```
But this is not blocking, as this is just cosmetic preference.

Reply:
Ok, i understood then. Already done